db/migrate/20210504150201_add_delivery_partner_to_csv_upload.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddDeliveryPartnerToCsvUpload < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :partnership_csv_uploads, :delivery_partner, index: true
```

Comment:
Foreign key?

Reply:
If the delivery partner / lead provider is optional, would a foreign key be required? It would have to be nullable ?

Reply:
And doesn't Rails just set the foreign key as the reference anyway? Not sure there is any difference between `add_reference :partnership_csv_uploads, :delivery_partner, index: true, null: false, foreign_key: true` and `add_reference :partnership_csv_uploads, :delivery_partner, index: true, null: false` at database level ?

Reply:
Have added a null: false to delivery_partner reference

Reply:
Although we do not have a catch if users jump the url and upload a file without a delivery_partner_id in the session

Reply:
And now cypress is failing for the above I suspect

Reply:
Will combine the features

Reply:
Funnily enough cypress runs locally

Reply:
Good to go now